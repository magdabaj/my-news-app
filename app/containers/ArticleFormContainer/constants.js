/*
 *
 * ArticleFormContainer constants
 *
 */

export const ADD_ARTICLE = "app/ArticleFormContainer/ADD_ARTICLE";
export const ADD_ARTICLE_SUCCESS = "app/ArticleFormContainer/ADD_ARTICLE_SUCCESS";
export const ADD_ARTICLE_ERROR = "app/ArticleFormContainer/ADD_ARTICLE_ERROR";
export const ADD_ARTICLE_CANCELED = "app/ArticleFormContainer/ADD_ARTICLE_CANCELED";

