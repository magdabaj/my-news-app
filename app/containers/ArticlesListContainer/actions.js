/*
 *
 * ArticlesListContainer actions
 *
 */

import {
  GET_ALL_ARTICLES,
  GET_ARTICLES_FOR_TAG,
  GET_ARTICLES_FOR_TAG_FAILED,
  GET_ARTICLES_FOR_TAG_SUCCESS, START_ADD
} from "./constants";

export const getAllArticles = () => ({
  type: GET_ALL_ARTICLES
});

export const getArticlesForTag = tagId => ({
  type: GET_ARTICLES_FOR_TAG,
  tagId,
});

export const getArticlesForTagSuccess = articles => ({
  type: GET_ARTICLES_FOR_TAG_SUCCESS,
  articles,
});

export const getArticlesForTagFailed = error => ({
  type: GET_ARTICLES_FOR_TAG_FAILED,
  error,
});

export const startAdd = tagName => ({
  type: START_ADD,
  tagName,
});