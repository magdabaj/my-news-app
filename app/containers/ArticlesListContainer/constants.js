/*
 *
 * ArticlesListContainer constants
 *
 */

export const GET_ALL_ARTICLES = "app/ArticlesListContainer/GET_ALL_ARTICLES";

export const GET_ARTICLES_FOR_TAG = "app/ArticlesListContainer/GET_ARTICLES_FOR_TAG";
export const GET_ARTICLES_FOR_TAG_SUCCESS = "app/ArticlesListContainer/GET_ARTICLES_FOR_TAG_SUCCESS";
export const GET_ARTICLES_FOR_TAG_FAILED = "app/ArticlesListContainer/export const GET_ARTICLES_FOR_TAG_FAILED";
