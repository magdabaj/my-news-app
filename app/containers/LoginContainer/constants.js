/*
 *
 * LoginContainer constants
 *
 */

export const LOGIN = "app/LoginContainer/LOGIN";
export const LOGIN_SUCCESS = "app/LoginContainer/LOGIN_SUCCESS";
export const LOGIN_FAILED = "app/LoginContainer/LOGIN_FAILED";

export const CANCEL_LOGIN = "app/LoginContainer/CANCEL_LOGIN";
