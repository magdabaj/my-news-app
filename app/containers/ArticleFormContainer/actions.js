/*
 *
 * ArticleFormContainer actions
 *
 */

import {ADD_ARTICLE, ADD_ARTICLE_CANCELED, ADD_ARTICLE_ERROR, ADD_ARTICLE_SUCCESS} from "./constants";

export const addArticle = (article, tagId, user) => ({
  type: ADD_ARTICLE,
  article,
  tagId,
  user
});

export const addArticleSuccess = article => ({
  type: ADD_ARTICLE_SUCCESS,
  article,
});

export const addArticleError = error => ({
  type: ADD_ARTICLE_ERROR,
  error,
});

export const addArticleCanceled = () => ({
  type: ADD_ARTICLE_CANCELED,
});