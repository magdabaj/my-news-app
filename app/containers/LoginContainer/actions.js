/*
 *
 * LoginContainer actions
 *
 */

import {CANCEL_LOGIN, LOGIN, LOGIN_FAILED, LOGIN_SUCCESS} from "./constants";

export const login = (email, password) => ({
  type: LOGIN,
  email,
  password,
});

export const loginSuccess = user => ({
  type: LOGIN_SUCCESS,
  user,
});

export const loginFailed = error => ({
  type: LOGIN_FAILED,
  error,
});

export const cancelLogin = () => ({
  type: CANCEL_LOGIN,
});