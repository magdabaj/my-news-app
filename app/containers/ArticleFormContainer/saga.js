import { take, call, put, select, all, takeLatest } from 'redux-saga/effects';
import {ADD_ARTICLE, ADD_ARTICLE_CANCELED} from "./constants";
import {addArticle} from "../../utils/api/articlesApi";
import {addArticleError, addArticleSuccess} from "./actions";
import {goBack} from "react-router-redux";

export function* addArticleSaga(action) {
    const {article, tagId, userId} = action;
    try {
        const createdArticle = yield call(addArticle, article, tagId, userId);
        yield put(addArticleSuccess(article));
    } catch (e) {
        yield put(addArticleError(e.message))
    }
}

export function* watchArticleAdd() {
    yield takeLatest(ADD_ARTICLE, addArticleSaga);
}

export function* watchAddCanceled() {
    yield takeLatest(ADD_ARTICLE_CANCELED, () => put(goBack()));
}

// Individual exports for testing
export default function* rootSaga() {
  yield all([
      watchArticleAdd(),
          watchAddCanceled()
      ])
}
