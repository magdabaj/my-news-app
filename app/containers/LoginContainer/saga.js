import { all, take, call, put, select, takeLatest } from 'redux-saga/effects';
import {cancelLogin, login, loginFailed, loginSuccess} from "./actions";
import {loginUser} from "../../utils/api/usersApi";
import {goBack} from "react-router-redux";
import {CANCEL_LOGIN, LOGIN} from "./constants";

export function* handleDone() {
    yield put(goBack());
}

export function* handleUserLogin(action) {
    const {email, password} = action;
    try{
        const user = yield call(loginUser,email, password);
        yield put(loginSuccess(user));
        yield put(goBack());
    } catch (e) {
        yield put(loginFailed(e.message))
    }
}

export function* doLoginSaga() {
    yield takeLatest(LOGIN, handleUserLogin);
}

export function* cancelSaga() {
    yield takeLatest(CANCEL_LOGIN, handleDone);
}

// Individual exports for testing
export default function* loginContainerSaga() {
  yield all([
      doLoginSaga(),
      cancelSaga(),
  ])
}
