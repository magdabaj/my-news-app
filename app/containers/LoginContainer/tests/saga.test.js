/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
// import { take, call, put, select } from 'redux-saga/effects';
// import loginContainerSaga from '../saga';

// const generator = loginContainerSaga();

import {doLoginSaga, handleUserLogin} from "../saga";
import {LOGIN} from "../constants";
import {put, takeLatest} from "redux-saga/effects";
import {loginFailed, loginSuccess} from "../actions";
import {goBack} from "react-router-redux";

const action = {
  type: LOGIN,
    email: 'email',
    password: 'password'
}

describe("loginContainerSaga Saga", () => {
  let loginUserGenerator;

  beforeEach(() => {
    loginUserGenerator = handleUserLogin({action})

    const callDescriptor = loginUserGenerator.next(action.email, action.password).value
    expect(callDescriptor).toMatchSnapshot()
  });
  it("should dispatch loginSuccess if it gets the data successfully", () => {
    const response = {
      email: 'email',
      password: 'password',
      token: 'token'
    }
    const loginUserSuccess = loginUserGenerator.next(response).value
    expect(loginUserSuccess).toEqual(put(loginSuccess(response)))
  });
  // it('should inject goBack action', () => {
  //   const goBackMocked = jest.fn()
  //   const goBackSuccess = loginUserGenerator.next(goBackMocked()).value
  //   expect(goBackSuccess).toEqual(put(goBackMocked()))
  // })
  it('should call the loginFailed action if logging fails', () => {
    const response = new Error('Some error')
    const loginUserFailed = loginUserGenerator.throw(response).value
    expect(loginUserFailed).toEqual(put(loginFailed(response.message)))
  })
});

describe('do login saga', () => {
  const watchLoginSaga = doLoginSaga();

  it('should start task to watch for LOGIN action', () => {
    const takeLatestDescriptor = watchLoginSaga.next().value
    expect(takeLatestDescriptor).toEqual(takeLatest(LOGIN, handleUserLogin))
  })
})