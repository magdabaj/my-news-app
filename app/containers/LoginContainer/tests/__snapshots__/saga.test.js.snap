// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginContainerSaga Saga Expect to have unit tests specified 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`loginContainerSaga Saga should call the loginFailed action if logging fails 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`loginContainerSaga Saga should dispatch loginSuccess if it gets the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`loginContainerSaga Saga should inject goBack action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
